### PR TITLE
[win32] re-work WinSystemWin32  + RenderSystemDX

### PR DESCRIPTION
**Pull request by afedchin**
In order to preparation for UWP I reworked Win32 windowing and rendering subsystems to make it possible to reuse dx code with minimal changes in uwp part.

### CODE REVIEW

**Comment by afedchin** [comment]
jenkins build this please

File: xbmc/guilib/Geometry.h [hunk 1]
```diff
@@ -273,8 +273,54 @@ template <typename T> class CRectGen
   }
 };
 
+template <typename T> struct SizeGen
```
Comment (FernetMenta) [inline xbmc/guilib/Geometry.h]: this entire block here is platform code, only used on Windows, right?

Comment (afedchin) [inline xbmc/guilib/Geometry.h]: right, currently it's uses on windows only

Comment (FernetMenta) [inline xbmc/guilib/Geometry.h]: why don't you move it to some windows only file? I think this is much better than those kone of workarounds: https://github.com/xbmc/xbmc/pull/12377/commits/17a9442bda35881ef11224bb2c8b3fad874ffee8

**Comment by afedchin** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by afedchin** [comment]
jenkins build this please

**Comment by afedchin** [comment]
jenkins build this please

**Comment by afedchin** [comment]
jenkins build this please

**Comment by Rechi** [comment]
@afedchin this needs a rebase

**Comment by afedchin** [comment]
jenkins build this please

**Comment by Rechi** [comment]
@afedchin is this ready or does it still need some work?

jenkins build this please

File: xbmc/windowing/windows/WinSystemWin32DX.cpp [hunk 2]
```diff
@@ -18,181 +18,323 @@
...
+    adapter++;
+  }
+  if (!deviceFound)
+    return;
+
+  CLog::Log(LOGDEBUG, __FUNCTION__": Hooking into UserModeDriver on device %S. ", displayDevice.DeviceKey);
+  wchar_t* keyName =
+#ifndef _M_X64
+  // on x64 system and x32 build use UserModeDriverNameWow key
+  CSysInfo::GetKernelBitness() == 64 ? keyName = L"UserModeDriverNameWow" :
```
Comment (Rechi) [inline xbmc/windowing/windows/WinSystemWin32DX.cpp]: `#include "utils/SystemInfo.h"` is missing

Comment (afedchin) [inline xbmc/windowing/windows/WinSystemWin32DX.cpp]: thanks.

**Comment by Rechi** [comment]
unrelated Android-ARM failure

**Comment by afedchin** [comment]
ok then. If this introduce new bugs, we will fix them.

**Pull request merged**